### ISSUE TITLE
Hugely increase mocking of tests

### ISSUE THREAD

**Issue by ikeydoherty** [ikeydoherty/clr-boot-manager#14]
TL;DR Get coverage up to 85%

**Comment by ikeydoherty** [comment]
2.0 should feature hugely enhanced mocking to raise the coverage of the codebase to a **minimum** of 80%.

UEFI and BIOS functionality should be tested in **all** bootloader configurations, with mocking of the bootloader UEFI spec for PartUUID detection in systemd-class bootloaders.

**Comment by ikeydoherty** [comment]
This is a **blocker** for 2.0

**Comment by ikeydoherty** [comment]
Tracking this in the new `legacy-boot` branch.

Prior to merge into master:

```
  lines......: 63.9% (1134 of 1774 lines)
  functions..: 78.4% (91 of 116 functions)
```

Removal of `euid` checks:
```
  lines......: 65.4% (1193 of 1825 lines)
  functions..: 74.2% (98 of 132 functions)
```

**Comment by ikeydoherty** [comment]
After introducing noop blkid harness
```
  lines......: 67.7% (1236 of 1825 lines)
  functions..: 80.3% (106 of 132 functions)
```

**Comment by ikeydoherty** [comment]
Damn but this is difficult. After moving to the system vtable, and "allowing" mounting:

```
Overall coverage rate:
  lines......: 68.6% (1258 of 1835 lines)
  functions..: 79.6% (109 of 137 functions)
```

**Comment by ikeydoherty** [comment]
Removing EFI_MIXED support in systemd class as they don't support it:

```
  lines......: 70.1% (1220 of 1740 lines)
  functions..: 79.7% (106 of 133 functions)
```

**Comment by ikeydoherty** [comment]
PR sent in https://github.com/ikeydoherty/clr-boot-manager/pull/39

**Comment by ikeydoherty** [comment]
Checking on the `coverage` branch

**Comment by ikeydoherty** [comment]
I'll happily mark this resolved when PR #46 goes in, even though it's not 85% on the code yet, it's 90%+ on the functions.

**Comment by ikeydoherty** [comment]
Marking this complete. Note the coverage must remain consistently high now.
